feat: Add app manager

Ytemiloluwa opened the pull request:
Hello @AryanCypherock this PR is now ready for review and you can run the tests with `poetry run pytest packages/app_manager/tests -vv`

Ytemiloluwa committed:
Add sdk manager utility functions

Ytemiloluwa committed:
Add auth_card & device proto files

Ytemiloluwa committed:
add types for proto

Ytemiloluwa committed:
Added auth card to services

Ytemiloluwa committed:
added mock files for app_manager

Ytemiloluwa committed:
Add app manager getlogs functions

Ytemiloluwa committed:
add app manager files

Ytemiloluwa committed:
Add tests for sdk manager

AryanCypherock commented on packages/app_manager/scripts/prebuild_auth.sh:
```diff
@@ -0,0 +1,132 @@
+@dataclass
+class AuthCardResponse:
+    response: Union[AuthCardSerialSigResponse, AuthCardChallengeSigResponse, None, AuthCardFlowCompleteResponse, None] = None
+EOF
```
why is this hardcoded? @Ytemiloluwa

Ytemiloluwa replied:
That is unused script for the auth card and auth device which was not needed for the python sdk according to muzzafar, I have removed it now.

Ytemiloluwa committed:
cleaned up the prebuild script

AryanCypherock commented:
# PR Review: Prebuild Scripts (app_manager and core)

## Summary

Prebuild flow is not reliable yet. The `prebuild.sh` scripts in `packages/app_manager` and `packages/core` don’t correctly generate `types.py` due to issues in the `extract_types` script, and there’s no automated way to run prebuild before building/running the project. Required setup steps are also not clearly documented.

## What’s wrong and how to fix

### 1) `extract_types` script is not configured correctly

- **Symptoms:** `types.py` is empty or missing expected enums/interfaces after running prebuild.
- **Likely causes:**
  - Enum extraction expects `@dataclass` but `betterproto` enums are declared as `class X(betterproto.Enum)` without decorators.
  - The consolidated `types.py` is missing imports (`dataclass`, `typing`, `betterproto`).
  - Dependency ordering of interfaces may not be respected, causing partial/empty output.
- **Fix direction:**
  - Update enum parsing to match `class <Name>(betterproto.Enum):` without relying on `@dataclass`.
  - Prepend the generated `types.py` with imports:
    - `from dataclasses import dataclass`
    - `from typing import List, Optional`
    - `import betterproto`
  - Ensure the script topologically sorts and emits interfaces after enums, and writes to the correct `types.py` path for each package.


### 2) `prebuild.sh` reliability and paths

- **Symptoms:** `protoc` runs but the plugin fails or output directories are incomplete; `types.py` not generated.
- **Likely causes:**
  - `betterproto` compiler plugin not installed.
  - Submodule protos not present (`git submodule update --init --recursive`).
  - Output folders not created before calling `protoc`.
  - Proto paths mismatch between packages (manager vs core protos).
- **Fix direction:**
  - Ensure `mkdir -p` for the output root before `protoc`.
  - Verify proto paths against `submodules/common/proto/...` for each package.
  - Confirm `protoc` and plugin availability locally before running the script; fail fast with a clear message if missing.

### Add these dependecies in poetry tool
- **Dependencies (runtime):**
    - Generated modules import `betterproto`, so keep it in main deps:
      ```toml
      [tool.poetry.dependencies]
      betterproto = "^1.2.5"
      ```
    - Or via command: `poetry add betterproto@^1.2.5`
  - **Dev dependencies (codegen):**
    - The compiler plugin is only needed for prebuild/codegen:
      ```toml
      [tool.poetry.group.dev.dependencies]
      betterproto = { version = "^1.2.5", extras = ["compiler"] }
      ```

### 3) No prebuild automation (manual-only)

- **Problem:** There’s no standard way to ensure prebuild runs before build/test/run.
- **Fix direction (pick one or more):**
  - Add a simple top-level `scripts/prebuild_all.sh` that calls both package scripts and document it as the first step.
  - Add a Makefile with `make build` that runs both prebuilds, and ask devs/CI to use it.
  - Add a CI job to run prebuild on PRs to keep generated files up-to-date.
  - Optionally add a lightweight Python wrapper (`scripts/prebuild.py`) that shells out to each package’s `prebuild.sh`.

### 4) Missing README setup for running prebuild

- **Problem:** New contributors won’t know what to install or how to run prebuild.
- **Add to README:**
  - **Prereqs (system + repo):**
    - Install Protocol Buffers compiler: `brew install protobuf` (macOS) or `sudo apt-get install protobuf-compiler` (Linux).
    - Initialize submodules: `git submodule update --init --recursive`.
  - **How to run:**
    - a documented top-level convenience script/Make target.
  - **When to run:** after clone/setup, after any proto changes, and before committing generated files.

## Acceptance criteria

- Running `packages/app_manager/scripts/prebuild.sh` generates non-empty `common.py`, `error.py`, `manager.py`, and a valid `types.py` in `packages/app_manager/src/proto/generated/`.
- Running `packages/core/scripts/prebuild.sh` generates non-empty `common.py`, `core.py`, `error.py`, and a valid `types.py` in `packages/core/src/encoders/proto/generated/`.
- README documents prerequisites and a single, repeatable prebuild command path (script or Make target).

Ytemiloluwa committed:
fix prebuild scripts and proto generated types

Ytemiloluwa commented:
Hello @AryanCypherock I have updated the extract_types script which now generates the `types.py` files for both core and app manager. I have moved the generated folders into git ignore as well. correct types are now written to the types.py files and I have added a makefile that runs both prebuilds. the `README` has been updated as well and all tests can be run with 
`poetry run pytest -v` 


<img width="1381" height="375" alt="Screenshot 2025-08-24 at 19 35 46" src="https://github.com/user-attachments/assets/c4d89972-449f-4193-b076-4636a6a5de1a" />

AryanCypherock commented on README.md:
```diff
@@ -50,6 +78,102 @@ packages/
+Run the prebuild process in these scenarios:
+- After cloning the repository for the first time
+- After pulling changes that include updates to `.proto` files
+- Before committing generated files
```
remove this line

Ytemiloluwa replied:
all 4 lines or line 119 only?

AryanCypherock commented on scripts/clean.sh:
this script removes node_modules, remove it or update it according to python project

Ytemiloluwa replied:
Okay

AryanCypherock commented on Makefile:
```diff
@@ -0,0 +1,54 @@
+
+# Clean generated files
+clean:
+	@echo "Cleaning generated files..."
+	find . -name "*.pyc" -delete
+	find . -name "__pycache__" -type d -exec rm -rf {} +
+	find . -name "*.egg-info" -type d -exec rm -rf {} +
+	find . -name "dist" -type d -exec rm -rf {} +
+	find . -name "build" -type d -exec rm -rf {} +
```
this deletes build folders from submodules as well, it should not

AryanCypherock commented on scripts/setupTurboCache.py:
This script was used to setup turboRepo cache, since we're not using turbo repo for this repository it is not needed.

Ytemiloluwa committed:
updated readme and makefile
